jgiven-html5-report/src/main/java/com/tngtech/jgiven/report/html5/Html5AttachmentGenerator.java

```diff
@@ -28,171 +13,249 @@
-            log.error( "Error while decoding the attachment to BufferedImage ", e );
+            BufferedImage before = ImageIO.read(new ByteArrayInputStream(imageBytes));
+            BufferedImage after = scaleDown(before);
+            base64thumb = bufferedImageToBase64(after, extension);
+        } catch (IOException e) {
+            log.error("Error while decoding the attachment to BufferedImage ", e);
         }
         return base64thumb;
     }
 
-    static BufferedImage scaleDown( BufferedImage before ) {
+    static BufferedImage scaleDown(BufferedImage before) {
         double xFactor = Math.min(1.0, MINIMAL_THUMBNAIL_SIZE / (double) before.getWidth());
         double yFactor = Math.min(1.0, MINIMAL_THUMBNAIL_SIZE / (double) before.getHeight());
 
         double factor = Math.max(xFactor, yFactor);
 
         int width = (int) Math.round(before.getWidth() * factor);
         int height = (int) Math.round(before.getHeight() * factor);
-        BufferedImage after = new BufferedImage( width, height, BufferedImage.TYPE_INT_ARGB );
+        BufferedImage after = new BufferedImage(width, height, BufferedImage.TYPE_INT_ARGB);
 
         AffineTransform at = new AffineTransform();
-        at.scale( factor, factor );
-        AffineTransformOp scaleOp = new AffineTransformOp( at, AffineTransformOp.TYPE_BILINEAR );
+        at.scale(factor, factor);
+        AffineTransformOp scaleOp = new AffineTransformOp(at, AffineTransformOp.TYPE_BILINEAR);
 
-        return scaleOp.filter( before, after );
+        return scaleOp.filter(before, after);
     }
 
-    private byte[] bufferedImageToBase64( BufferedImage bi, String extension ) {
+    private byte[] bufferedImageToBase64(BufferedImage bi, String extension) {
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
         byte[] imageArray = {};
         try {
-            ImageIO.write( bi, extension, baos );
+            ImageIO.write(bi, extension, baos);
             imageArray = baos.toByteArray();
             baos.close();
-        } catch( IOException e ) {
-            log.error( "Error while decoding the compressed BufferedImage to base64 ", e );
+        } catch (IOException e) {
+            log.error("Error while decoding the compressed BufferedImage to base64 ", e);
         }
         return imageArray;
     }
+
+    private void writeThumbnailForSVG(File initialSVG, File thumbnailSVG) {
+        String base64PNGImage = getPNGFromSVG(initialSVG);
+        byte[] scaledDownInBytes = compressToThumbnail(base64PNGImage, "png");
+        Dimension imageDimension = getImageDimension(scaledDownInBytes);
+        String base64ScaledDownContent = BaseEncoding.base64().encode(scaledDownInBytes);
+        createSVGThumbFile(thumbnailSVG, base64ScaledDownContent, imageDimension);
+    }
+
+    Dimension getImageDimension(byte[] givenImage) {
+        Dimension dimension = new Dimension();
+        try (ByteArrayInputStream bais = new ByteArrayInputStream(givenImage)) {
```

Comment:
It's just a minor thing, but could we maybe come up with a better naming here? Maybe just something like "inputstream" instead of "bias".